### DISCUSSION TITLE
Translatable and Filament\Forms\Components\Builder

### DISCUSSION THREAD

**Discussion by PepijnVDK1** [filamentphp/filament#5035] [Help]
Wondering if this is possible:
I'm having trouble implementing Spatie translatable combined with Filament\Forms\Components\Builder. 

Adding blocks to the builder does not add the language prefix (in the JSON) . This might be due to the fact that the builder
already returns a JSON object. 

For what it's worth, 
the code I have is: 
`                    Builder::make('blocks')
                        ->collapsible()
                        ->blocks(PageManager::getPageBlocks())
                        ->columnSpan(2),`

and in the model:
`public $translatable = ['title', 'blocks'];`
`    protected $casts = [
        'blocks' => 'array',
    ];`

What am I missing?
Did anyone have a similar issue and maybe found a solution?

Thanks in advance

**Comment by KaanTanis** [comment]
Same problem

**Reply by PepijnVDK1** [reply]
I found a way to do it, but it's hacky though:

protected function mutateFormDataBeforeSave(array $data): array
    {
        $data['blocks'] = json_encode($data['blocks']);

        return $data;
    }

    protected function mutateFormDataBeforeFill(array $data): array
    {
        $data['blocks'] = json_decode($data['blocks'], true);

        return $data;
    }

This converts the json data before saving to a string. And when loading the resource it reverses it.
I'm doing it this way until there is a better solution
Hope this helps you out for now


Edit: I have no idea why my code block is this weird
